src/ghc.jl

```diff
@@ -0,0 +1,98 @@
+"""
+    GHC(k, variogram)
+
+Geostatistical Hierarchical Clustering.
+
+Long description....
```

Comment:
It would be nice to have a clear description of the method before continuing with the implementation. You should be able to list the parameters of the method and explain to the end user what they do.